packages/propel/src/charts/radar-chart/root.tsx

```
@@ -0,0 +1,81 @@
+import { getLegendProps } from '../components/legend';
+import { CustomXAxisTick } from '../components/tick';
+
+const RadarChart = <T extends string, K extends string>(props: TRadarChartProps<T, K>) => {
+  const { data, radars, dataKey, margin, showTooltip, legend, className, angleAxis } = props;
+
+  // states
+  const [activeIndex, setActiveIndex] = useState<number | null>(null);
+  const [activeLegend, setActiveLegend] = useState<string | null>(null);
+
+  const itemKeys = useMemo(() => radars.map((radar) => radar.key), [radars]);
+  const itemLabels = useMemo(() => radars.reduce((acc, radar) => ({ ...acc, [radar.key]: radar.name }), {}), [radars]);
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)